### PR TITLE
eth/catalyst: error on nil withdrawals post-shanghai

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
According to the spec, we need to return an error if withdrawals are set before shanghai and if they are set to nil after shanghai

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
@mkalinin I think this is currently underspecified. Should we return an error if the withdrawals are nil instead of empty list post?
Should we do that post-shanghai or always on fcuV2?

File: eth/catalyst/api.go [hunk 1]
```diff
@@ -164,6 +164,9 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update beacon.ForkchoiceStateV1, pa
 
 // ForkchoiceUpdatedV2 is equivalent to V1 with the addition of withdrawals in the payload attributes.
 func (api *ConsensusAPI) ForkchoiceUpdatedV2(update beacon.ForkchoiceStateV1, payloadAttributes *beacon.PayloadAttributes) (beacon.ForkChoiceResponse, error) {
+	if payloadAttributes != nil && payloadAttributes.Withdrawals == nil {
+		return beacon.STATUS_INVALID, fmt.Errorf("withdrawals have to be set to empty slice, not nil")
```
Comment (fjl) [inline eth/catalyst/api.go:168]:
```suggestion
		return beacon.STATUS_INVALID, errors.New("missing withdrawals list")
```

**Comment by MariusVanDerWijden** [comment]
don't merge I need to fix something

**Comment by fjl** [comment]
Regarding the latest change to reject withdrawals before Shanghai in the engine API, I wonder why the miner does not catch this issue. It would be nicer to rely on the check in `FinalizeAndAssemble` instead of checking for the fork explicitly in the API handler.

**Review by fjl — APPROVED** [review]

**Pull request merged**